frontends/parsers/parserDriver.h

```diff
@@ -115,6 +115,12 @@ class P4ParserDriver final : public AbstractParserDriver {
     static const IR::P4Program *parse(FILE *in, std::string_view sourceFile,
                                       unsigned sourceLine = 1);
 
+    static std::pair<const IR::P4Program *, const Util::InputSources *> parseProgramSources(
```

Comment:
Please add an explanation here when to use this method (as opposed to `parse`. For both methods.